### PR TITLE
CORGI-591: Make manifests use filter for released components

### PR DESCRIPTION
**Pull request by juspence**
@mprpic No rush to look at this. I checked some OpenShift data in the API and saw the filter appeared to be working properly, so I've added the same filtering to our ProductStream manifest templates.

### CODE REVIEW

File: corgi/core/files.py [hunk 1]
```diff
@@ -67,11 +67,12 @@ class ProductManifestFile(ManifestFile):
     def render_content(self) -> str:
 
         latest_components = self.obj.get_latest_components()  # type: ignore[attr-defined]
```
Comment (mprpic) [inline corgi/core/files.py:69]: Is this unused now then?

**Review by mprpic — APPROVED** [review]
LGTM!

File: corgi/core/files.py (hunk 1 above)
Comment (juspence) [reply corgi/core/files.py:69]: No, see the line below it. We need to get all the latest components in a stream, then find which of those are released. We can't do the other way around - find all the components that are released, then take their latest builds / versions - as a side-effect of how this filter was implemented. Refactoring is planned.

Comment (mprpic) [reply corgi/core/files.py:69]: Ah, fail, I didn't realize released_ was being run on top of latest_. Ignore me! :)

**Pull request merged**